### ISSUE TITLE
Sentry Transactions are missing

### ISSUE THREAD

**Issue by mpass99** [openHPI/poseidon#428]
We have gained the impression that Sentry Transactions are missing. Especially, the transaction of the WebSocket connection is missing often.

**Comment by mpass99** [comment]
TLDR: See `codeocean-terraform!138`

When excluding the Sentry Relay (by re-configuring the Sentry Endpoint of Poseidon and CodeOcean to `ingest.sentry.io`) all transaction are received [[1]](https://codeocean.sentry.io/performance/codeocean:2dadd52af5be49e4931fab46faead398) [[2]](https://codeocean.sentry.io/performance/codeocean:28e82523739b45c7814e2c632df102f0) [[3]](https://codeocean.sentry.io/performance/codeocean:a4a4efdbf7cd400f92cf615068866fa3) [[4]](https://codeocean.sentry.io/performance/codeocean:3b231536d71c45b58313d812f495c11e) [[5]](https://codeocean.sentry.io/performance/codeocean:1f426232342f4b239fe5ec716812062f) [[6]](https://codeocean.sentry.io/performance/codeocean:b57c282939f445f78f918c297191d55c) [[7]](https://codeocean.sentry.io/performance/codeocean:d8c08353e3d149b89388c7fbefaadf24) [[8]](https://codeocean.sentry.io/performance/codeocean:11cb0954aaaf48bea793d763de48ba63) [[9]](https://codeocean.sentry.io/performance/codeocean:1f41adb546be45479ec6582e0c42f0b8) [[10]](https://codeocean.sentry.io/performance/codeocean:60230aa3184c4c858ef88c30717effef).

When using the Sentry Relay all transaction are received except for the transaction of the WebSocket connection. This transaction is received only in about 10% of the executions [[11]](https://codeocean.sentry.io/performance/codeocean:ce93ae07e07a444f9f44d8c03ac51063) [[12]](https://codeocean.sentry.io/performance/codeocean:32ad5102da58400db4d1ba48ebe6d7f1) [[13]](https://codeocean.sentry.io/performance/codeocean:d643602fc25e4914a54a59597428525e) [**[14]**](https://codeocean.sentry.io/performance/codeocean:88e241b4467a456ea0b1dfac8880667b) [[15]](https://codeocean.sentry.io/performance/codeocean:558f9ae46ba64f1eb3b51d8a65d201c8) [[16]](https://codeocean.sentry.io/performance/codeocean:d395768623f14c169d2660f782878516) [[17]](https://codeocean.sentry.io/performance/codeocean:0c64b3d597304e66af103d417a59d6a0) [[18]](https://codeocean.sentry.io/performance/codeocean:17f0b535ea4a4b4a8064e4831c0acef4) [[19]](https://codeocean.sentry.io/performance/codeocean:778f12d641bd45b196c12fa382d8318a) [[20]](https://codeocean.sentry.io/performance/codeocean:8558097754b4421d870d9e647612e1fe).

Therefore, we can limit the error to the Sentry Relay. When having a look in the Sentry Relay logs, we have no warnings or errors, but for each transaction the Sentry relay logs the following lines:
``` log
2023-08-23T11:02:01.234479Z DEBUG request{method=POST uri=/api/5667283/envelope/ version=HTTP/1.0}: tower_http::trace::on_request: started processing request
2023-08-23T11:02:01.236683Z TRACE request{method=POST uri=/api/5667283/envelope/ version=HTTP/1.0}: relay_server::endpoints::common: queueing envelope
2023-08-23T11:02:01.236752Z DEBUG request{method=POST uri=/api/5667283/envelope/ version=HTTP/1.0}: tower_http::trace::on_response: finished processing request latency=2 ms status=200
2023-08-23T11:02:01.238480Z TRACE relay_server::actors::processor: processing json transaction
2023-08-23T11:02:01.244494Z TRACE relay_server::actors::envelopes: sending envelope to sentry endpoint
```

The only difference is that for the one successful execution, the log contains 6 transactions, and for the other executions just 5 ([AllTransactions](https://github.com/openHPI/poseidon/files/12418590/AllTransactions.txt) - [WebSocketMissing](https://github.com/openHPI/poseidon/files/12418591/WebSocketMissing.txt)). However, [the Nginx Log](https://github.com/openHPI/poseidon/files/12418641/Nginx.txt) shows that we receive 6 requests for either. The difference: For the cases where the WebSocket Transaction is missing one request has the StatusCode 413 `Payload Too Large`.

Therefore, it's not the fault of the Sentry Relay this time, but our configuration of the Nginx `client_max_body_size` (or that we send too much data in our Poseidon spans).

**Comment by MrSerth** [comment]
Awesome, thanks for identifying. I merged the branch but haven't deployed the change yet.

**Comment by MrSerth** [comment]
I have deployed the change for nginx, so that we can observe it now.

**Comment by MrSerth** [comment]
I checked the Sentry transactions and indeed was able to confirm that the newer transactions are complete. Hence, I am closing this issue.
